usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -696,6 +708,75 @@
+    LEFT OUTER JOIN (
+    SELECT
+        faba.award_id,
+        TO_JSON(
+            SORT_ARRAY(
+                COLLECT_SET(
+                    NAMED_STRUCT(
+                        'id', fa.id,
+                        'account_title', fa.account_title,
+                        'federal_account_code', fa.federal_account_code
+                    )
+                )
+            )
+        ) AS federal_accounts,
+        COLLECT_SET(
+        DISTINCT CONCAT(
+            'agency=', COALESCE(agency.toptier_code, ''),
+            'faaid=', COALESCE(fa.agency_identifier, ''),
+            'famain=', COALESCE(fa.main_account_code, ''),
+            'aid=', COALESCE(taa.agency_id, ''),
+            'main=', COALESCE(taa.main_account_code, ''),
+            'ata=', COALESCE(taa.allocation_transfer_agency_id, ''),
+            'sub=', COALESCE(taa.sub_account_code, ''),
+            'bpoa=', COALESCE(taa.beginning_period_of_availability, ''),
+            'epoa=', COALESCE(taa.ending_period_of_availability, ''),
+            'a=', COALESCE(taa.availability_type_code, '')
+        )
+        ) AS tas_paths,
+        COLLECT_SET(
+        CONCAT(
+            'aid=', COALESCE(taa.agency_id, ''),
+            'main=', COALESCE(taa.main_account_code, ''),
+            'ata=', COALESCE(taa.allocation_transfer_agency_id, ''),
+            'sub=', COALESCE(taa.sub_account_code, ''),
+            'bpoa=', COALESCE(taa.beginning_period_of_availability, ''),
+            'epoa=', COALESCE(taa.ending_period_of_availability, ''),
+            'a=', COALESCE(taa.availability_type_code, '')
+        )
+        ) AS tas_components,
+        -- "CASE" put in place so that Spark value matches Postgres; can most likely be refactored out in the future
+        CASE
+            WHEN SIZE(COLLECT_SET(faba.disaster_emergency_fund_code)) > 0
+                THEN SORT_ARRAY(COLLECT_SET(faba.disaster_emergency_fund_code))
+            ELSE NULL
+        END AS disaster_emergency_fund_codes,
+        COLLECT_SET(taa.treasury_account_identifier) AS treasury_account_identifiers,
+        COLLECT_SET(
+            TO_JSON(
+                NAMED_STRUCT(
+                    'name', UPPER(rpa.program_activity_name),
+                    'code', LPAD(rpa.program_activity_code, 4, "0")
+                )
+            )
+        ) AS program_activities
+    FROM
+        global_temp.treasury_appropriation_account taa
+    INNER JOIN int.financial_accounts_by_awards faba ON (taa.treasury_account_identifier = faba.treasury_account_id)
+    INNER JOIN global_temp.federal_account fa ON (taa.federal_account_id = fa.id)
+    INNER JOIN global_temp.toptier_agency agency ON (fa.parent_toptier_agency_id = agency.toptier_agency_id)
+    LEFT JOIN global_temp.ref_program_activity rpa ON (faba.program_activity_id = rpa.id)
+    WHERE
+        faba.award_id IS NOT NULL
+    GROUP BY
+        faba.award_id
+    ) TREASURY_ACCT ON (TREASURY_ACCT.award_id = a.id)
```

Comment:
Does this have to live on `subaward_search`? Currently the View that powers the Elasticsearch Subaward Index joins `subaward_search` and `award_search`, which would allow you to reference the fields from this join in the subaward view. This is what I am referring to: https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/database_scripts/etl/subaward_es_view.sql#L118